[AI Store] Remove `webmozart/assert` library

Spomky opened the pull request:
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no <!-- please update src/**/CHANGELOG.md files -->
| Docs?         | no <!-- required for new features -->
| Issues        | none
| License       | MIT

Hi,

I am not sure w need the `webmozart/assert` dependency for just one verification.
Let me squash this PR before any action (I am not with my usual setup at the moment).

Regards.

Nyholm reviewed (changes requested):
Thank you. I agree

chr-hertel commented:
yes, agree as well, but for the Platform component you'd need to replace more code - or keep it for now and only take action on store

Spomky commented:
Yes indeed. I will continue this PR when I get back home on Thursday.

chr-hertel commented on src/platform/src/Bridge/Azure/OpenAI/EmbeddingsModelClient.php:
```diff
@@ -34,11 +33,11 @@ public function __construct(
-        Assert::stringNotEmpty($deployment, 'The deployment must not be empty.');
-        Assert::stringNotEmpty($apiVersion, 'The API version must not be empty.');
-        Assert::stringNotEmpty($apiKey, 'The API key must not be empty.');
+        !str_starts_with($this->baseUrl, 'http://') || throw new \InvalidArgumentException('The base URL must not contain the protocol.');
+        !str_starts_with($this->baseUrl, 'https://') || throw new \InvalidArgumentException('The base URL must not contain the protocol.');
+        '' !== $deployment || throw new \InvalidArgumentException('The deployment must not be empty.');
+        '' !== $apiVersion || throw new \InvalidArgumentException('The API version must not be empty.');
+        '' !== $apiKey || throw new \InvalidArgumentException('The API key must not be empty.');
```
Let's us the component specific exception here please: `Symfony\AI\Platform\Exception\InvalidArgumentException`

Spomky replied:
Indeed I had missed this class.
I have updated it wherever necessary.

OskarStark commented on src/platform/src/Bridge/Azure/OpenAI/EmbeddingsModelClient.php:
```diff
@@ -34,11 +34,11 @@ public function __construct(
-        Assert::stringNotEmpty($deployment, 'The deployment must not be empty.');
-        Assert::stringNotEmpty($apiVersion, 'The API version must not be empty.');
-        Assert::stringNotEmpty($apiKey, 'The API key must not be empty.');
+        !str_starts_with($this->baseUrl, 'http://') || throw new InvalidArgumentException('The base URL must not contain the protocol.');
+        !str_starts_with($this->baseUrl, 'https://') || throw new InvalidArgumentException('The base URL must not contain the protocol.');
```
we can use an `OR` here

OskarStark commented:
I will prepare a PR to add unit tests for the code parts using the lib. So we can merge my PR first, then rebase and check the CI 👍

OskarStark commented:
Here you go:
* https://github.com/symfony/ai/pull/126

chr-hertel commented:
Merged Oskar's tests, rebase please :)

Spomky commented:
Thanks. Rebased, squashed and tests are ok.
I noted there are `InvalidArgumentException` classes in `ai-platform`, `agent` and `store`.
As `agent` and `store` depend `on ai-platform`, do you think interesting to remove redundant classes?
